python/pylibcugraph/pylibcugraph/graphs.pyx

```diff
@@ -330,8 +330,12 @@ cdef class MGGraph(_GPUGraph):
         assert_CAI_type(weight_array, "weight_array", True)
         if edge_id_array is not None:
             assert_CAI_type(edge_id_array, "edge_id_array")
+            if len(edge_id_array) != len(src_array):
+                raise ValueError('Edge id array must be same length as edgelist')
         if edge_type_array is not None:
-            assert_CAI_type(edge_type_array, "edge_type_array")        
+            assert_CAI_type(edge_type_array, "edge_type_array")
```

Comment:
For this and others, if you pass `True` it will allow None for this check, which has the advantage of checking that the value is either a CAI type or None, but nothing else. (see `weight_array` above)
```suggestion
        assert_CAI_type(edge_type_array, "edge_type_array", True)
```